source/Symbol/SwiftASTContext.cpp

```diff
@@ -6152,6 +6152,17 @@ bool SwiftASTContext::IsFixedSize(CompilerType compiler_type) {
   return false;
 }
 
+const llvm::fltSemantics &
+SwiftASTContext::GetFloatTypeSemantics(size_t byte_size) {
+  const size_t bit_size = byte_size * 8;
```

Comment:
This is highly suspect; IIRC Float80 is stored as 16 bytes.

Reply:
@jrose-apple, IIRC, the FP80 is backed by Builtin.FPIEEE80. so, the size should be correct?

Reply:
@compnerd Depends quite a bit on how `byte_size` is derived in the caller. Where is this function used?

Reply:
@stephentyrone As far as I can tell, GetFloatTypeSemantics is only used in https://github.com/llvm/llvm-project/blob/master/lldb/source/Core/DumpDataExtractor.cpp

Reply:
https://github.com/llvm/llvm-project/blob/master/lldb/source/Core/DumpDataExtractor.cpp#L573-L574

The backing for floating point should be 10-bytes (FP80) on x87 and 8-bytes or 16-bytes for single and double precision.

Reply:
Also, https://github.com/llvm/llvm-project/blob/master/lldb/source/Core/DumpDataExtractor.cpp#L561-L562 - we will never really come down this path.  Why?!  We should just move forward with this for now I think.  This is effectively dead code, but has reasonable behaviour.  Whenever someone comes around to adding support for this in Swift, we should be able to catch this.

CC: @xiaobai @dcci

Reply:
If this is indeed dead code can we add an llvm_unreachable instead?

Reply:
@JDevlieghere - it is only dead for the time being.  I don't expect this to be dead forever, this should be something that should get implemented at some point - we would want to be able to render `Float` and `Double` in the Swift context properly.

Reply:
@JDevlieghere done

Reply:
@compnerd Of course and when that time comes we should implement it correctly. I prefer the unreachable because it conveys the right signal.